lib/AST/ASTContext.cpp

```diff
@@ -1442,11 +1443,10 @@ void ASTContext::addSearchPath(StringRef searchPath, bool isFramework,
 
 void ASTContext::addModuleLoader(std::unique_ptr<ModuleLoader> loader,
                                  bool IsClang) {
-  if (IsClang) {
-    assert(!getImpl().TheClangModuleLoader && "Already have a Clang module loader");
+  if (IsClang && !getImpl().TheClangModuleLoader)
```

Comment:
Rather than change this, I think you just shouldn't pass IsClang. Changing the name of the parameter might make more sense.

Reply:
I'm not sure what you mean. Are you suggesting to dyn_cast<ClangModuleLoader> here?

```
void ASTContext::addModuleLoader(std::unique_ptr<ModuleLoader> loader) {
  if (auto *clangModuleLoader = dyn_cast<ClangModuleLoader>(loader.get())
    if (!getImpl().TheClangModuleLoader)
      getImpl().TheClangModuleLoader = clangModuleLoader;

  getImpl().ModuleLoaders.push_back(std::move(loader));
}
```

Reply:
Sorry, I meant that the *call* to `addModuleLoader` for the DWARFModuleLoader should just pass `false` (the default).